mmpose/datasets/pipelines/mesh_transform.py

```diff
@@ -0,0 +1,399 @@
+
+        joints_3d_visible_flipped[left, :] = joints_3d_visible[right, :]
+        joints_3d_visible_flipped[right, :] = joints_3d_visible[left, :]
+
+    # Flip horizontally
+    joints_3d_flipped[:, 0] = -joints_3d_flipped[:, 0]
+    joints_3d_flipped = joints_3d_flipped * joints_3d_visible_flipped
+
+    return joints_3d_flipped, joints_3d_visible_flipped
+
+
+@PIPELINES.register_module()
+class LoadIUVFromFile(object):
+    """Loading IUV image from file."""
+
+    def __init__(self, to_float32=False):
+        self.to_float32 = to_float32
+        self.color_type = 'color'
+        # channel relations: iuv->bgr
+        self.channel_order = 'bgr'
+
+    def __call__(self, results):
+        """Loading image from file."""
+        has_iuv = results['has_iuv']
+        use_iuv = results['ann_info']['use_IUV']
+        if has_iuv and use_iuv:
+            iuv_file = results['iuv_file']
+            iuv = mmcv.imread(iuv_file, self.color_type, self.channel_order)
+            if iuv is None:
+                raise ValueError('Fail to read {}'.format(iuv_file))
+        else:
+            has_iuv = 0
+            iuv = None
+
+        results['has_iuv'] = has_iuv
+        results['iuv'] = iuv
+        return results
+
+
+@PIPELINES.register_module()
+class IUVToTensor():
+    """Transform IUV image to part index mask and uv coordinates image. The 3
+    channels of IUV image means: part index, u coordinates, v coordinates.
+
+    Required key: 'iuv', 'ann_info'.
+    Modifies key: 'part_index', 'uv_coordinates'.
+
+    Args:
+        results (dict): contain all information about training.
+    """
+
+    def __call__(self, results):
+        iuv = results['iuv']
+        if iuv is None:
+            H, W = results['ann_info']['iuv_size']
+            part_index = torch.zeros([1, H, W], dtype=torch.long)
+            uv_coordinates = torch.zeros([2, H, W], dtype=torch.float32)
+        else:
+            part_index = torch.LongTensor(iuv[:, :, 0])[None, :, :]
+            uv_coordinates = torch.FloatTensor(iuv[:, :, 1:]) / 255
+            uv_coordinates = uv_coordinates.permute(2, 0, 1)
+        results['part_index'] = part_index
+        results['uv_coordinates'] = uv_coordinates
+        return results
+
+
+@PIPELINES.register_module()
+class MeshRandomChannelNoise():
+    """Data augmentation with random channel noise.
+
+    Required keys: 'img'
+    Modifies key: 'img'
+
+    Args:
+        noise_factor (float): Multiply each channel with
+         a factor between``[1-scale_factor, 1+scale_factor]``
+    """
+
+    def __init__(self, noise_factor=0.4):
+        self.noise_factor = noise_factor
+
+    def __call__(self, results):
+        """Perform data augmentation with random channel noise."""
+        img = results['img']
+
+        # Each channel is multiplied with a number
+        # in the area [1-self.noise_factor, 1+self.noise_factor]
+        pn = np.random.uniform(1 - self.noise_factor, 1 + self.noise_factor, 3)
```

Comment:
```python
pn = np.random.uniform(1 - self.noise_factor, 1 + self.noise_factor, (1, 3))
img = cv2.multiply(img, pn)
```

if you want speed and allow inplace change of img, then change the second line to
```python
cv2.multiply(img, pn, img)
```

Reply:
OK, I will modify the code.